kratos/processes/find_global_nodal_elemental_neighbours_process.h

```diff
@@ -0,0 +1,449 @@
+
+                    for(auto& item : recv_map[color])
+                    {
+                        auto& recv_node = mr_model_part.Nodes()[item.first]; 
+                        auto& neighbours = recv_node.GetValue(NEIGHBOUR_ELEMENTS);
+                        for(auto& gp : item.second.GetContainer())
+                            neighbours.push_back(gp);
+                    }
+                }
+            } //after this loop is finished neighbours are ok for the owner nodes
+
+            //fill back the recv_map with the updated information
+            for(int color : colors)
+            {
+                if(color >= 0)
+                {
+                    for(auto& item : recv_map[color])
+                    {
+                        //item.first contains the id of the node
+                        //item.second contains the list of neighbours
+                        auto& recv_node = mr_model_part.Nodes()[item.first]; 
+                        item.second = recv_node.GetValue(NEIGHBOUR_ELEMENTS);
+                    }
+
+                    //obtain the final list of neighbours for nodes owned on color
+                    auto final_gp_map = SendRecv(recv_map[color], color, color );
+
+                    //update the local database
+                    for(auto& item : final_gp_map)
+                    {
+                        auto& recv_node = mr_model_part.Nodes()[item.first]; 
+                        recv_node.GetValue(NEIGHBOUR_ELEMENTS) = item.second;
+                    }
+                }
+            }
+        }
+
+        auto constructor_functor =  Kratos::ComputeNeighbourListFunctor<
+                            ModelPart::NodesContainerType, 
+                            Variable< GlobalPointersVector< Element > >
+                            > (rNodes, NEIGHBOUR_ELEMENTS);
+
+        GlobalPointerCommunicator<Element> pointer_comm(mrComm, constructor_functor );
+        auto id_proxy = pointer_comm.Apply(
+                [](GlobalPointer<Element>& gp){return gp->Id();}
+        );
+
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(mr_model_part.Nodes().size()); ++i)
+        {
+            auto it = mr_model_part.NodesBegin() + i;
+            auto& neighbours = it->GetValue(NEIGHBOUR_ELEMENTS);
+            neighbours.shrink_to_fit();
+            std::sort(neighbours.ptr_begin(), neighbours.ptr_end(),
+                [&id_proxy](GlobalPointer<Element>& gp1, GlobalPointer<Element>& gp2)
+                {
+                    return id_proxy.Get(gp1) < id_proxy.Get(gp2);
+                }
+            );
+        }
+    
+    }
+
+    void ClearNeighbours()
+    {
+        NodesContainerType& rNodes = mr_model_part.Nodes();
+        for(NodesContainerType::iterator in = rNodes.begin(); in!=rNodes.end(); in++)
+        {
+            auto& rN = in->GetValue(NEIGHBOUR_ELEMENTS);
+            rN.erase(rN.begin(),rN.end() );
```

Comment:
Why don't use `GlobalPointerVector::Clear()`?

Reply:
Clear is not doing shrink_to_fit, and here is important to actually free the memory